qa/tasks/cephfs/test_multifs.py

```diff
@@ -0,0 +1,397 @@
+                          'fsid={cephfs2_id}',
+                'osdcap': 'allow rw tag cephfs data={cephfs1_name}, allow rw tag cephfs ' + \
+                          'data={cephfs2_name}',
+                'cephfs_mntpt': None},
+            # mds cap without path and with r permission
+            {
+                'mdscap': 'allow r fsid={cephfs1_id}, allow r ' + \
+                          'fsid={cephfs2_id}',
+                'osdcap': 'allow r tag cephfs data={cephfs1_name}, allow r tag cephfs ' + \
+                          'data={cephfs2_name}',
+                'cephfs_mntpt': None},
+
+            # mds cap with path and with rw permission
+            {
+                'mdscap': 'allow rw fsid={cephfs1_id} path=/{cephfs_mntpt}, allow rw ' + \
+                          'fsid={cephfs2_id} path=/{cephfs_mntpt}',
+                'osdcap': 'allow rw tag cephfs data={cephfs1_name}, allow rw tag cephfs ' + \
+                          'data={cephfs2_name}',
+                'cephfs_mntpt': 'dir1'},
+
+            # mds cap with path and with r permission
+            {
+                'mdscap': 'allow r fsid={cephfs1_id} path=/{cephfs_mntpt}, allow r ' + \
+                          'fsid={cephfs2_id} path=/{cephfs_mntpt}',
+                'osdcap': 'allow rw tag cephfs data={cephfs1_name}, allow rw tag cephfs ' + \
+                          'data={cephfs2_name}',
+                'cephfs_mntpt': 'dir1'},
+
+            # misbehaves
+            #{'mdscap': 'allow rw *', 'osdcap': 'allow rw tag cephfs data=*',
+            # 'cephfs_mntpt': '/'},
+            #{'mdscap': 'allow r *', 'osdcap': 'allow r tag cephfs data=*',
+            # 'cephfs_mntpt': '/'},
+        )
+
+        testcases_passed_count = 0
+        for testcase in testcases:
+            try:
+                self.run_test_auth_for_multiple_fss(**testcase)
+            except Exception as e:
+                print(e)
+                print('=' * 79)
+                print(str(testcases_passed_count) + ' test cases passed '
+                      'successfully.')
+                print('testcase details -')
+                print('mdscap = ' + testcase['mdscap'])
+                print('osdcap = ' + testcase['osdcap'])
+                if isinstance(testcase['cephfs_mntpt'], str):
+                    print('cephfs_mntpt = ' + testcase['cephfs_mntpt'])
+                else:
+                    print('cephfs_mntpt is {}'.format(testcase['cephfs_mntpt']))
+                print('=' * 79)
+                raise
+
+            self.run_test_auth_for_multiple_fss_resetup()
+            testcases_passed_count = testcases_passed_count + 1
+
+        print('=' * 79)
+        print('testcases_passed_count = ' + str(++testcases_passed_count))
+        print('total testcases = ' + str(len(testcases)))
+        print('=' * 79)
+
+    def run_test_auth_for_multiple_fss(self, mdscap, osdcap, cephfs_mntpt):
+        moncap = 'allow r'
```

Comment:
Where do you test for changes in MON caps to restrict access to a filesystem, 'allow r fsid=<fsid>' or 'allow rw fsid=<fsid>'

Reply:
Those are being added. I'll push those in some time.

Reply:
The tests have been added. All of them work fine with kernel as well as with FUSE.